frb_codegen/src/binary/commands.rs

```diff
@@ -194,6 +199,20 @@ pub(crate) struct IntegrateCommandArgs {
 
     #[clap(flatten)]
     pub common: CreateOrIntegrateCommandCommonArgs,
+
+    /// The template type to use for integration. This should usually match the type of flutter project
+    /// being integrating with.
+    #[clap(short, long, value_enum, default_value = "app")]
+    pub template: ProjectTypeArg
+}
+
+#[derive(Debug,Clone,ValueEnum)]
+pub(crate) enum ProjectTypeArg {
+    /// (default) a Flutter application
+    App,
+    /// A shareable Flutter project containing an API in Dart code with a 
+    /// platform-specific implementation
```

Comment:
nit: not really "platform-specific" indeed, since we can have same rust on different platforms (thus the text copied from official flutter may need a bit of change)